src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs

```diff
@@ -290,8 +303,12 @@ private void OnInventoryReceived(IInventory inventory)
             if (inventory is Transaction transaction)
                 system.Consensus?.Tell(transaction);
             system.Blockchain.Tell(inventory, ActorRefs.NoSender);
-            pendingKnownHashes.Remove(inventory.Hash);
-            knownHashes.Add(inventory.Hash);
```

Comment:
Why?

Reply:
This code is moved to func RenewKnownHashes

Reply:
Why move?

Reply:
Because of concurrent transaction handling. In this PR, transaction is verified in different threads. Renew knowhashes in different threads might not be a good idea, so I move the knowhashes renewing part to somewhere before transaction verification.

Reply:
I think we shouldn't use multithreading in actors. One actor one thread is good.

Reply:
That's in an ideal network topology, but in fact, most transactions actually come from one or several actors, which lead to low resource utilization.

Reply:
If most transactions come from one actor, I think a single node should not consume all our resources. And this situation is highly likely to attack.

Reply:
I think this might also be caused by special network topology, i.e. whole network can be seperated into several parts and these parts are connected to each other via only several or even one nodes. Transaction traffic via these nodes can be very busy, and these nodes might act similiar to attackers.

Reply:
> i.e. whole network can be seperated into several parts and these parts are connected to each other via only several or even one nodes

If so, it means that the network is not good. How to improve network connectivity is another topic, and I think the practice of not using multithreading in actors should not be easily broken.

Reply:
No, in normal case, it'll may also "seperated into several parts", due to network delay, the order of nodes, etc. But we'll remove multithreading, at least better than before.

Reply:
> > i.e. whole network can be seperated into several parts and these parts are connected to each other via only several or even one nodes
> 
> If so, it means that the network is not good. How to improve network connectivity is another topic, and I think the practice of not using multithreading in actors should not be easily broken.

I switched multithreading mode to actor pool, please have a look:)

Reply:
After testing the TPS effect of actor pool is the same as Qiao-Jin#39 and lower than original logic.